docker/Dockerfile.firedrake-um2n

```diff
@@ -0,0 +1,17 @@
+# Dockerfile for installing Firedrake plus UM2N and its dependencies with CPU support
+
+# Based on firedrake
+FROM firedrakeproject/firedrake:2024-10
```

Comment:
Just out of interest: why do you use this version? This also installs quite a few packages in `firedrake/src` that we don't need (FEMlium, Irksome, defcon, g-adopt, asQ, fascd, gusto, icepack, ngsPETSc), as well as `torch` which is not suitable for our needs.

Why not use `firedrakeproject/firedrake-vanilla:latest`?

Reply:
Ah good point, addressed in aad2d94af6e1db087ffbf3d2bb7d813831d689d3. This must've been something I forgot to update from the original implementation.